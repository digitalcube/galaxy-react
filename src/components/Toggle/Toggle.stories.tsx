import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Toggle, ToggleProps } from './Toggle';

const meta: Meta = {
  title: 'Components/Toggle',
  component: Toggle,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<ToggleProps> = (args) => <Toggle {...args} />;

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  checked: true,
};
