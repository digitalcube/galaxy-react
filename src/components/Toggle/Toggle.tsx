import React, { FC } from 'react';

export type ToggleProps = {
  checked: boolean;
  id: string;
  label: string;
  onChange: React.ChangeEventHandler<HTMLInputElement>;
} & Omit<
  React.DetailedHTMLProps<
    React.InputHTMLAttributes<HTMLInputElement>,
    HTMLInputElement
  >,
  'onChange' | 'type' | 'value' | 'checked' | 'className' | 'id'
>;

export const Toggle: FC<ToggleProps> = ({
  checked,
  onChange,
  id,
  label,
  ...inputProps
}) => {
  return (
    <div className="custom-control custom-switch c-toggle-switch">
      <input
        type="checkbox"
        className="custom-control-input"
        id={id}
        checked={checked}
        onChange={onChange}
        {...inputProps}
      />
      <label className="custom-control-label" htmlFor={id}>
        <span className="d-none">{label}</span>
      </label>
    </div>
  );
};
