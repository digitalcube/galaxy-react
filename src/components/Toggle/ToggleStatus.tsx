import React, { FC, PropsWithChildren } from 'react';
import { useMemo } from 'react';

export type ToggleStatusProps = PropsWithChildren<{
  className?: string;
}>;

export const ToggleStatus: FC<ToggleStatusProps> = ({
  className,
  children,
}) => {
  const classNames = useMemo(() => {
    return ['toggle-status-box', className].filter(Boolean);
  }, [className]);
  return <div className={classNames.join(' ')}>{children}</div>;
};
