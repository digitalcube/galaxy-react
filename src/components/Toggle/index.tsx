export * from './Toggle';
export * from './ToggleStatus';
