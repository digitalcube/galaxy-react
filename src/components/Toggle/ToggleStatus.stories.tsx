import React from 'react';
import { Meta, Story } from '@storybook/react';
import { ToggleStatus, ToggleStatusProps } from './ToggleStatus';
import { Toggle } from './Toggle';

const meta: Meta = {
  title: 'Components/ToggleStatus',
  component: ToggleStatus,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<ToggleStatusProps> = (args) => (
  <ToggleStatus {...args}>
    <Toggle
      checked={true}
      id="dummy-toggle"
      onChange={() => undefined}
      label="dummy"
      status={{
        isActive: true,
        fontWeight: 'bold',
        inactiveLabel: 'inactiveLabel',
        activeLabel: 'activeLabel',
      }}
    />
  </ToggleStatus>
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {};
