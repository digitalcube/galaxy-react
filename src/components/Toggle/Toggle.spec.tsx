import { render } from '@testing-library/react';
import React from 'react';
import { Default as Toggle } from './Toggle.stories';

describe('Toggle', () => {
  it('to match snapshot(default)', () => {
    const { asFragment } = render(<Toggle {...(Toggle.args as any)} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
