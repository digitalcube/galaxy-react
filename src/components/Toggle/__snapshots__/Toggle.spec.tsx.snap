// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle to match snapshot(default) 1`] = `
<DocumentFragment>
  <div
    class="custom-control custom-switch c-toggle-switch"
  >
    <input
      checked=""
      class="custom-control-input"
      type="checkbox"
    />
    <label
      class="custom-control-label"
    >
      <span
        class="d-none"
      />
    </label>
  </div>
</DocumentFragment>
`;
